onnxruntime/core/providers/cpu/signal/utils.h

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/framework/tensor.h"
+
+namespace onnxruntime {
+namespace signal {
+
+template <typename T>
+static T get_scalar_value_from_tensor(const Tensor* tensor) {
+  ORT_ENFORCE(tensor->Shape().Size() == 1, "ratio input should have a single value.");
+
+  auto data_type = tensor->DataType()->AsPrimitiveDataType()->GetDataType();
+  switch (data_type) {
+    case ONNX_NAMESPACE::TensorProto_DataType_FLOAT:
+      return static_cast<T>(*reinterpret_cast<const float*>(tensor->DataRaw()));
+    case ONNX_NAMESPACE::TensorProto_DataType_DOUBLE:
+      return static_cast<T>(*reinterpret_cast<const double*>(tensor->DataRaw()));
```

Comment:
>(tensor->DataRaw())); [](http://example.com/codeflow?start=60&length=21)

What is the reason of using DataRaw() vs Data<T>()?

Reply:
I wrote the code originally ~1 yr ago... I don't recall the difference now...

Reply:
I would simply write return static_cast<T>(*tensor->Data<float>()).  Or whatever type it needs to be. Casting feels misplaced, kind of suppressing the type.

Reply:
Not entirely sure if this is right, but I tried to use `Data()` and it seems a lot simpler. PTAL.

Reply:
Data<>()

Reply:
Sorry, are you saying that what I have written is wrong? It compiles as-is.
Could you please explain?

Reply:
instead of  

`return static_cast<T>(*reinterpret_cast<const float*>(tensor->DataRaw()));`

I am suggesting 

`return static_cast<T>(*tensor->Data<float>());`

Reply:
The code I currently have is `return *tensor->Data<T>();` which seems clear, avoids casts, and makes the compiler happy. LMK if there's some reason to prefer your suggestion instead.

Reply:
You need static_cast. Data<T>() does no casts, it returns the type that is stored and it actually checks for that, otherwise an exception is thrown. So you want to request the correct type, then cast.